collections/specprocessor/standalone_scripts/imagerotator.php

```diff
@@ -21,8 +21,9 @@
 $rotateManager->setRotateMode($rotateMode);
 $rotateManager->setMsgOutMode($msgMode);
 
-if($targetPath) $rotateManager->batchRotateImages($targetPath);
-else echo 'ERROR: target path is empty';
+$targetPath = realpath($PUBLIC_IMAGE_UPLOAD_ROOT . $reqPath);
```

Comment:
@egbot, my understanding is that this file is probably only used by you and maybe a handful of other power users. As previously written, essentially any file path could be used, and then a bash command is run on the path. I kind of implemented a whitelisted subset of paths under the assumptions that any file requiring this code was located in $PUBLIC_IMAGE_UPLOAD_ROOT. If this is not the intended case, we should definitely restrict the paths that can be accessed by this code.

Reply:
🤔